test/parallel/test-net-persistent-keepalive.js

```diff
@@ -0,0 +1,34 @@
+var common = require('../common');
+var assert = require('assert');
+var net = require('net');
+
+var serverConnection;
+var echoServer = net.createServer(function(connection) {
+  serverConnection = connection;
+  connection.setTimeout(0);
+  assert.notEqual(connection.setKeepAlive, undefined);
+  connection.on('end', function() {
+    connection.end();
+  });
+});
+echoServer.listen(common.PORT);
+
+echoServer.on('listening', function() {
+  var clientConnection = new net.Socket({});
+  // send a keepalive packet after 1000 ms
+  // and make sure it persists
+  clientConnection.setKeepAlive(true, 1000);
+  clientConnection.connect(common.PORT);
+  clientConnection.setTimeout(0);
+  assert.equal(clientConnection._setKeepAlive[0], true);
+  assert.equal(clientConnection._setKeepAlive[1], 1000);
+
+  setTimeout(function() {
+    // make sure both connections are still open
+    assert.equal(serverConnection.readyState, 'open');
+    assert.equal(clientConnection.readyState, 'open');
+    serverConnection.end();
+    clientConnection.end();
+    echoServer.close();
+  }, 1200);
```

Comment:
Is it possible to lower the timeout?  Tests should ideally finish in less than a second.